source/retryable-writes/retryable-writes.rst

```diff
@@ -229,6 +228,8 @@ user.
 Supported Write Operations
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+MongoDB 3.6 will support retryability for some, but not all, write operations.
```

Comment:
(This comment is for lines 223-226 but github won't let me add it there).

I'm wondering why we don't raise the original error when the retry fails.

Reply:
https://github.com/mongodb/specifications/pull/174/commits/bf0a17b573b0a7f97ef00b9b88c274736a056590 clarifies that "fails" refers to a network error

Reply:
> If the server selected for a retry attempt does not support retryable writes (e.g. mixed-version cluster), retrying is not possible and drivers MUST raise the original network error to the user.

@jyemin: the spec language above and corresponding pseudo-code doesn't address a server selection failure. Based on the current pseudo-code, I expect that `selectServer("writable")` before the retry attempt would throw an exception if it fails to return a writable server. I'd agree that failing to select a writable server for the retry attempt is not much different (from the user's perspective) as a selected server not support retryable writes, so throwing the original network exception works for me.

Reply:
Let's leave it as is.